phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/skin/ApplicationSidebarToggleGroupSkin.java

```diff
@@ -44,7 +47,10 @@ protected ToggleButton convertToToggleButton(CategoryDTO category) {
         final ToggleButton categoryButton = createSidebarToggleButton(category.getName());
 
         categoryButton.setId(String.format("%sButton", category.getId().toLowerCase()));
-        categoryButton.setOnAction(event -> getControl().getOnCategorySelection().accept(category));
+        categoryButton.setOnAction(event -> {
+            getControl().setSelectedElement(category);
+            getControl().getOnCategorySelection().accept(category);
```

Comment:
Wouldn't it make sense to combine these two calls?

Reply:
I have a better idea:
Why don't we remove the callback methods from the sidebar toggle groups and the search box?
Instead we can subscribe directly to the properties for changes.
This makes it also much easier to add listeners where they are needed.

Reply:
My problem with this is that you would need to know the internal implementation (i.e. the properties). With the call back, this is transparent.

Reply:
I'm not sure what you mean?
Don't you have the same problem with the callback methods?
If you don't know them you can't use them, it's the same with properties.

The properties are quite easy to use. You can just add your callback/listener to the property by calling `Property#addListener`

Reply:
You're right.